### DISCUSSION TITLE
The Torch version is overwritten at each startup

### DISCUSSION THREAD

**Discussion by neurogen-dev** [vladmandic/sdnext#687] [Q&A]
I install Torch 2.1 CUDA12.1 nightly build, then I run Automatic.
During startup it starts to install Torch 2.0 Cuda 11.8, overwriting my installation. 

How can I avoid this?

**Comment by vladmandic** [comment]
the only reason it would get replaced is it cannot find a valid version of any of three packages: torch torchaudio torchvision
so if you install torch, but don't install torchaudio or torchvision, it will force install all three from known combo.

otherwise, you can just uninstall torch, specify environment variable with torch you want and let installer do its job:
for example, this is default for nvidia gpus:
`set TORCH_COMMAND="torch torchaudio torchvision --index-url https://download.pytorch.org/whl/cu118"`
